src/PrestaShopBundle/Controller/Admin/AuthorizationController.php

```diff
@@ -0,0 +1,333 @@
+        }
+
+        return $this->renderLoginPage($request, [
+            'loginForm' => $loginForm->createView(),
+            'showLoginForm' => false,
+            'forgotPasswordForm' => $forgotPasswordForm->createView(),
+            'showForgotPasswordForm' => $showForgotPasswordForm,
+        ]);
+    }
+
+    /**
+     * Handle password resetting.
+     *
+     * @DemoRestricted(redirectRoute="_admin_login")
+     *
+     * @param Request $request
+     * @param int $employeeId
+     * @param string $resetToken generated password reset token
+     *
+     * @return Response
+     */
+    public function resetPasswordAction(Request $request, $employeeId, $resetToken)
+    {
+        try {
+            /** @var PasswordResettingEmployee $employee */
+            $employee = $this->getQueryBus()->handle(new GetEmployeeForPasswordReset((int) $employeeId));
+        } catch (DomainException $e) {
+            $this->addError($this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('_admin_login');
+        }
+
+        $resetPasswordForm = $this->createForm(ResetPasswordType::class, null, [
+            'email' => $employee->getEmail()->getValue(),
+        ]);
+        $resetPasswordForm->handleRequest($request);
+
+        if ($resetPasswordForm->isSubmitted() && $resetPasswordForm->isValid()) {
+            try {
+                $this->getCommandBus()->handle(new ResetPasswordCommand(
+                    $employee->getEmployeeId()->getValue(),
+                    $employee->getEmail()->getValue(),
+                    $resetToken,
+                    $resetPasswordForm->getData()['reset_password']
+                ));
+
+                $this->addSuccess($this->trans(
+                    'Your password has been changed successfully.',
+                    'Admin.Login.Notification'
+                ));
+
+                return $this->redirectToRoute('_admin_login');
+            } catch (EmployeeNotFoundException $e) {
+                $this->addError($this->trans(
+                    'An error occurred while attempting to reset your password.',
+                    'Admin.Login.Notification'
+                ));
+            } catch (DomainException $e) {
+                $this->addError($this->getErrorMessageForException($e, $this->getErrorMessages()));
+            }
+        }
+
+        return $this->renderLoginPage($request, [
+            'showLoginForm' => false,
+            'showForgotPasswordForm' => false,
+            'showResetPasswordForm' => true,
+            'resetPasswordForm' => $resetPasswordForm->createView(),
+            'employeeId' => $employeeId,
+            'resetToken' => $resetToken,
+        ]);
+    }
+
+    /**
+     * Render the login page.
+     *
+     * @param Request $request
+     * @param array $templateVars
+     *
+     * @return Response
+     */
+    private function renderLoginPage(Request $request, array $templateVars = [])
```

Comment:
This render function seems a bit too complicated, and I think the main reason is that there is only one twig template used for every use cases The template should be split into dedicated templates for each use cases rather that managing many `if` conditions in this method **AND** in the template
For example I'm pretty sure the resetPassword form can be managed completely separately but it's juste an example

Reply:
This would also allow to simplify the `messages.html.twig` template which is used for every case as well This doesn't help understanding nor overriding